app/javascript/src/components/Editor.tsx

```diff
@@ -118,6 +122,40 @@ class Editor extends React.Component<EditorProps, EditorState> {
     this.model = model
   }
 
+  private handleSceneName = (
+    currentState: any,
+    newStory: any,
+    ) => {
+    let currentNodes = [...currentState.story.nodes]
+    let updatedName: string = ''
+    let latestLink = Object.assign(new DefaultLinkModel(), currentState.story.links[currentState.story.links.length - 1])
+    let sourcePort = Object.assign(new DefaultPortModel(false, ''))
+    let targetNode = Object.assign(new DefaultNodeModel())
+
+    currentNodes.forEach(node => {
+      if(node.id === latestLink.source) {
+        sourcePort = node.ports.filter((port: DefaultPortModel) => port.id === latestLink.sourcePort)
+        updatedName = sourcePort[0].label
+      }
+
+      if(node.id === latestLink.target) {
+         targetNode = node
+      }
+    })
+
+    if(targetNode.name === "New Scene") {
+      let targetNodeIndex = currentNodes.indexOf(targetNode)
+
+      targetNode.name = updatedName
+
+      currentNodes.splice(targetNodeIndex, 1, targetNode)
```

Comment:
what does this line do? replacing the targetNode with a newer copy of itself?

Reply:
Yeah! Since the targetNode has been updated (the name), it needs to be added in somewhere. This line will just replace that particular node with the new one

Reply:
I think I would have expected that updating the `targetNode` object (which is a reference to an object in the array of `currentNodes`) would have been all you need to do to push the change into the ledger. I'm guessing that's not the case though if you have this line and the 2 below it as well.